src/OpenRiaServices.Client/Framework/OperationBase.cs

```diff
@@ -22,19 +23,27 @@ public abstract class OperationBase : INotifyPropertyChanged
         private bool _isErrorHandled;
         private readonly CancellationTokenSource _cancellationTokenSource;
 
+        private protected static TaskScheduler CurrentSynchronizationContextTaskScheduler => SynchronizationContext.Current != null ? TaskScheduler.FromCurrentSynchronizationContext() : TaskScheduler.Default;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="OperationBase"/> class.
         /// </summary>
         /// <param name="userState">Optional user state.</param>
         /// <param name="supportCancellation"><c>true</c> to setup cancellationTokenSource and use that to handle cancellation</param>
         protected OperationBase(object userState, bool supportCancellation)
+            : this(userState, supportCancellation ? new CancellationTokenSource() : null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OperationBase"/> class.
+        /// </summary>
+        /// <param name="userState">Optional user state.</param>
+        /// <param name="cancellationTokenSource">cancellationTokenSource to use for cancellation</param>
+        protected OperationBase(object userState, CancellationTokenSource cancellationTokenSource)
```

Comment:
How is the disposal of the CancellationTokenSource handled? I Can't find any Dispose in this class. Since it's not always passed in by the external code it might not get disposed properly